lib/dnd.js

```diff
@@ -312,6 +312,15 @@ Dnd.prototype._children = function (d, parent, children) {
   return children
 }
 
+/*
+ * A traveling node is `illegal` if it's not droppable (based on the user defined droppable callback) or
+ * its parent is a transient node.
+ */
+Dnd.prototype._isIllegal = function (source, newParent) {
+  return !this.tree.options.droppable.call(this.tree, this.tree.nodes[source.id], this.tree.nodes[newParent && newParent.id]) ||
```

Comment:
This is minor, but I would have put the check against the transient ID first since you know for sure that test is cheap, whereas the cost of the call to droppable is unknown since it comes from outside the library